client/src/js/components/bhCurrencySelect.js

```diff
@@ -97,8 +97,6 @@ function bhCurrencySelect($scope, Currencies) {
 
     // if the two array lengths are equal, it means every currency is disabled
     $ctrl.allDisabled = ($ctrl.currencies.length === array.length);
-    if ($ctrl.allDisabled) {
-      $ctrl.form.$setValidity('currencies', false);
-    }
+    $ctrl.form.currency.$setValidity('currency', !$ctrl.allDisabled);
```

Comment:
Also, this moves from changing the global form validity to changing the input's validity.  There was some weird behavior when I attempted to use the FormController as an ngModelController, so I am simply using the one input's controller in this case.